### PR TITLE
NullHandler: dataframe schema and metadata mismatch now raises exception

### PR DESCRIPTION
**Pull request by guderkar**

### CODE REVIEW

File: src/featurestorebundle/delta/feature/NullHandler.py [hunk 1]
```diff
@@ -16,9 +20,11 @@ def fill_nulls(self, df: DataFrame, feature_list: FeatureList) -> DataFrame:
 
         return df.fillna(fill_dict)
 
-    def to_storage_format(self, df: DataFrame, feature_list: FeatureList) -> DataFrame:
+    def to_storage_format(self, df: DataFrame, feature_list: FeatureList, entity: Entity) -> DataFrame:
+        self.__features_validator.validate(entity, df, feature_list)
```
Comment (ds-lukaslangr) [inline src/featurestorebundle/delta/feature/NullHandler.py:24]: can it even happen that the metadata don't fit data during write?

File: src/featurestorebundle/delta/feature/NullHandlerTest.py [hunk 2]
```diff
@@ -175,14 +175,34 @@ def test_conversion(self):
...
+
+        input_feature_list = FeatureList(
+            [
+                FeatureInstance(
+                    self.__entity.name, "f1", "this is feature 1", "string", {}, FeatureTemplate("f1", "this is feature 1", "", "str")
+                ),
+            ]
+        )
+
+        with self.assertRaisesRegex(Exception, "Dataframe columns do not match declared features"):
```
Comment (ds-lukaslangr) [inline src/featurestorebundle/delta/feature/NullHandlerTest.py:203]: RaisesRegex?

File: src/featurestorebundle/delta/feature/NullHandler.py (hunk 1 above)
Comment (guderkar) [reply src/featurestorebundle/delta/feature/NullHandler.py:24]: I don't think so but better be safe than sorry

File: src/featurestorebundle/delta/feature/FeaturesPreparer.py [hunk 3]
```diff
@@ -25,7 +25,7 @@ def __init__(
     def prepare(self, features_storage: FeaturesStorage) -> DataFrame:
         features_data = self.__features_joiner.join(features_storage)
         features_data = self.__null_handler.fill_nulls(features_data, features_storage.feature_list)
-        features_data = self.__null_handler.to_storage_format(features_data, features_storage.feature_list)
+        features_data = self.__null_handler.to_storage_format(features_data, features_storage.feature_list, features_storage.entity)
```
[the remark this answers is not in the document]
Comment (guderkar) [reply src/featurestorebundle/delta/feature/FeaturesPreparer.py:28]: Maybe yes but then `to_storage_format` and `from_storage_format` would have different "interface" which does not seem right to me

File: src/featurestorebundle/delta/feature/NullHandlerTest.py (hunk 2 above)
Comment (guderkar) [reply src/featurestorebundle/delta/feature/NullHandlerTest.py:203]: Yes, besides it asserts Exception raise, it also checks for Exception message. There is no other method to check for exact message.

**Pull request merged**